emcc.py

```diff
@@ -1603,6 +1603,11 @@ def save_intermediate(name=None, suffix='js'):
         link_opts = [] if debug_level >= 4 or shared.Settings.CYBERDWARF else ['-strip-debug'] # remove LLVM debug if we are not asked for it
         if not shared.Settings.ASSERTIONS:
           link_opts += ['-disable-verify']
+        else:
+          # when verifying, LLVM debug info has some tricky linking aspects, and llvm-link will
+          # disable the type map in that case. we added linking to opt, so we need to do
+          # something similar, which we can do with a param to opt
+          link_opts += ['-disable-debug-info-type-map']
```

Comment:
Yeah, it's just a few lines in opt that add in linking capability. This weird debug info issue is the first wrinkle we've had with it really...

And yeah, we need to disable this optimization when we use upstream opt, that's done [here](https://github.com/kripken/emscripten/blob/master/emcc.py#L1516) (could probably be improved, we could check for vanilla directly now, I think there is some way)